### DISCUSSION TITLE
Drone Flip

### DISCUSSION THREAD

**Discussion by jonas-kupferschmid** [bitcraze/discussions#700] [Q&A]
Hello, I am still a beginner but want to create a series of youtube tutorials to help fellow beginners be able to get started with programming the Crazyflie. But I require some technical help with it. Thank you in advance for helping out!

I was able to perform a drone flip on a crazyflie for an ETH course, however a custom firmware was used there (for us to implement our own controller and functions). So I am having a hard time reproducing it with the official firmware.
For the tutorial, I wanted to make a drone flip when pressing a button on the cfclient Flight control page. For that, I wanted to make an open loop flip, by manually setting the thrusts. What is the best way to have the program (potentially in the APP layer, if not doable, then in firmware is also fine) run at a certain frequency where I can set the thrusts of each motor without getting overwritten? Is it possible to write an API and how can I do that?
Can I use the following existing params?
stabilizer.stop (is this enough, I saw that there are multiple controlers PID kalman etc, and default chooses 0 (auto), do i have to set one specific and turn off that specific controller?)
motorPowerSet.enable
motorPowerSet.m1-4 (what would the max value be that the motor can handle? when I set it)

**Comment by ataffanel** [comment]
Hi,

I have no experience implementing flipping on the Crazyflie, but depending of what the timing needs are you should be able to do it with the `motorPowerSet.enable` and `motorPowerSet.m1-4` parameters from the ground.

If done in the client you will likely be able to spawn a new Python thread that set parameters and uses sleep for timing. This would allow to do some very crude open-loop control.

A more proper solution might of course be to do the same in an app, triggered by a single parameter set.

I guess the actual proper way would be to implement the functionality in the stabilizing loop together with an app doing the high-level control.

Maybe for your purpose of making tutorials, all 3 methods makes sense to do in sequence, to show that things can be first hacked/experimented with form the ground and then gradually implemented in the firmware.

**Reply by bitdeckai** [reply]
Can it implement flipping on the Crazyflie now?

**Reply by gemenerik** [reply]
I replied in your new thread: https://github.com/orgs/bitcraze/discussions/1907
